pkg/storage/local.go

```diff
@@ -1032,7 +1041,7 @@ func (is *ImageStoreLocal) FinishBlobUpload(repo, uuid string, body io.Reader, d
 
 	dst := is.BlobPath(repo, dstDigest)
 
-	if is.dedupe && is.cache != nil {
+	if is.dedupe && fmt.Sprintf("%v", is.cache) != fmt.Sprintf("%v", nil) {
```

Comment:
what if we use a switch where we use type check for each implementation we know we have, this time the only implementation is dynamodatabase, if it's not that type assume cache is nil

later when adding more databases we add check for them.

Thoughts?